CONTRIBUTING.md

```diff
@@ -1,60 +1,145 @@
+  * Use an existing resource (e.g. `internal/elasticsearch/security/system_user`) as a template.
+  * Some resources use the deprecated Terraform SDK, so only resources using the new Terraform Framework should be used as reference.
+* Use the generated API clients to interact with the Kibana APIs. (See [Working with Generated API Clients](#working-with-generated-api-clients)
+* Add documentation and examples for the resource. Update the generated docs with `make docs-generate`.
+* Write unit and acceptance tests.
 
 
-## Update documentation
+## Using Copilot
 
-Update documentation templates in `./templates` directory and re-generate docs via:
-```bash
-make docs-generate
-```
+GitHub Copilot can speed up development, but you’re responsible for correctness:
+* Create an issue describing the desired change and acceptance criteria.
+* Assign the issue to Copilot and iterate with prompts.
+* Review outputs carefully; add tests and adjust as needed.
+* Example issue: https://github.com/elastic/terraform-provider-elasticstack/issues/1219
 
-## Update `./CHANGELOG.md`
+### Useful Commands
 
-List of previous commits is a good example of what should be included in the changelog.
+* `make build`: Build the provider.
+* `make install`: Install the provider.
+* `make fmt`: Format the code.
+* `make lint`: Lint the code.
+* `make test`: Run unit tests.
+* `make docs-generate`: Generate documentation.
 
+#### Running Acceptance Tests
 
-## Pull request
+Acceptance tests spin up Elasticsearch, Kibana, and Fleet with Docker and run tests in a Go container.
 
-Format the code before pushing:
+Quick start (default stack version from Makefile):
 ```bash
-make fmt
+make docker-testacc
 ```
 
-Check if the linting:
+Run a single test with terraform debug enabled:
 ```bash
-make lint
+env TF_LOG=DEBUG make docker-testacc TESTARGS='-run ^TestAccResourceDataStreamLifecycle$$'
 ```
 
-Create a PR and check acceptance test matrix is green.
-
-## Run provider with local terraform
-
-TBD
+A way to forward debug logs to a file:
+```bash
+env TF_ACC_LOG_PATH=/tmp/tf.log TF_ACC_LOG=DEBUG TF_LOG=DEBUG make docker-testacc
+```
 
-## Releasing
+### Working with Generated API Clients
+
+If your work involves the Kibana API, the API client can be generated directly from the Kibana OpenAPI specs:
+- For Kibana APIs, use the generated client in `generated/kbapi`.
+- To add new endpoints, see [generated/kbapi/README.md](generated/kbapi/README.md).
+- Regenerate clients with:
+  ```sh
+  make transform generate
+  ```
+
+The codebase includes a number of deprecated clients which should not be used anymore:
+- `libs/go-kibana-rest`: Fork of an external library, which is not maintained anymore.
+- `generated/alerting`
+- `generated/connectors`
+- `generated/slo`
+- `internal/clients/*`: Manually written clients. These should only be used/extended if it is not possible to use the generated clients.
```

Comment:
```suggestion
```

This isn't really true. Much of this package configures one of the Stack clients, providing endpoints/credentials/etc from the provider configuration to the client. 

There's some code in here which is used as a bridge between the provider resources and the older Kibana/Elasticsearch clients. We don't need to continue with that as much, but there are still cases where it may make sense, for instance in Kibana connectors have a bunch of non-trivial logic managing API defaults which kind of makes sense in a client layer. 

I don't think there's a need for a hard rule on not making additions here.

Reply:
True, I didn't realize this also contains all the wrapper code for the generated client. I'll remove that entry.